src/main/java/org/apache/sling/scriptingbundle/plugin/processor/FileProcessor.java

```
@@ -118,81 +123,86 @@ void processRequiresFile(@NotNull Path requiresFile,
+                boolean isOptional = aQute.bnd.osgi.Constants.OPTIONAL.equals(requiresParameter.getValue().get(aQute.bnd.osgi.Constants.RESOLUTION_DIRECTIVE));
                 RequiredResourceTypeCapability.Builder requiredBuilder =
                         RequiredResourceTypeCapability.builder().withResourceType(resourceType);
+                if (isOptional) {
+                    requiredBuilder.withIsOptional();
+                }
                 extractVersionRange(requiresFile, requiredBuilder, requiresParameter.getValue().getVersion());
                 requiredCapabilities.add(requiredBuilder.build());
             }
         } catch (IOException e) {
-            log.error(String.format("Unable to read file %s.", requiresFile.toString()), e);
+            throw new UncheckedIOException(String.format("Unable to read file %s.", requiresFile.toString()), e);
         }
     }
 
     public void processScriptFile(@NotNull Path resourceTypeDirectory, @NotNull Path scriptPath,
                                    @NotNull ResourceType resourceType, @NotNull Set<ProvidedResourceTypeCapability> providedCapabilities) {
-        String filePath = scriptPath.toString();
-        String extension = FilenameUtils.getExtension(filePath);
-        if (StringUtils.isNotEmpty(extension) && scriptEngineMappings.containsKey(extension)) {
-            Path scriptFile = scriptPath.getFileName();
-            if (scriptFile != null) {
-                Path relativeResourceTypeFolder = resourceTypeDirectory.relativize(scriptPath);
-                int pathSegments = relativeResourceTypeFolder.getNameCount();
-                LinkedHashSet<String> selectors = new LinkedHashSet<>();
-                if (pathSegments > 1) {
-                    for (int i = 0; i < pathSegments - 1; i++) {
-                        selectors.add(relativeResourceTypeFolder.getName(i).toString());
-                    }
+        Path scriptFile = scriptPath.getFileName();
```

What was the point of refactoring this chunk? It doesn't seem to be related to the issue.

---

Initially I thought that we should also throw an exception if no engine was found for an extension. This was too strict though, but it is important to always log a WARN to make users aware of potential misconfigurations. I only removed the condition of line 140 because of this. I am gonna restore the condition for the empty extension to make it more visible.

---

Done in https://github.com/apache/sling-scriptingbundle-maven-plugin/pull/4/commits/4d6b9c5c26e17f21c5ef2b40072423f31151dc53

---

Thanks! Looks better now.